components/parcels/Foreclosure.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, useEffect } from "react";
+import { BigNumber } from "ethers";
+import { gql, useQuery } from "@apollo/client";
+import { Container } from "react-bootstrap";
+import { Framework } from "@superfluid-finance/sdk-core";
+import type { Point } from "@turf/turf";
+import * as turf from "@turf/turf";
+import { GeoWebContent } from "@geo-web/content";
+import { Contracts } from "@geo-web/sdk/dist/contract/types";
+import { PCOLicenseDiamondFactory } from "@geo-web/sdk/dist/contract/index";
+import { MAX_LIST_SIZE, Parcel, ParcelsQuery } from "./ParcelList";
+import ParcelTable from "./ParcelTable";
+import { STATE } from "../Map";
+import { getParcelContent } from "../../lib/utils";
+
+interface ForeclosureProps {
+  sfFramework: Framework;
+  geoWebContent: GeoWebContent;
+  registryContract: Contracts["registryDiamondContract"];
+  setSelectedParcelId: React.Dispatch<React.SetStateAction<string>>;
+  setInteractionState: React.Dispatch<React.SetStateAction<STATE>>;
+  handleCloseModal: () => void;
+  setParcelNavigationCenter: React.Dispatch<React.SetStateAction<Point | null>>;
+  shouldRefetch: boolean;
+  setShouldRefetch: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+type Bid = Parcel & {
+  timestamp: number;
+  licenseOwner: string;
+};
+
+const foreclosureQuery = gql`
+  query Foreclosure($skip: Int) {
+    geoWebParcels(
+      first: 1000
```

Comment:
Do we need to get 1000 parcels here? Or can we limit it to `MAX_LIST_SIZE`?

Reply:
Here and in `NeedsTransfer` querying for only 25 parcels wouldn't result in 25 parcels foreclosed or that needs transfer because we don't have a way to query the subgraph directly for that, but e.g. need to call `isPayerBidActive` on each to tell if a parcel is foreclosed.

So we just get a pool of parcels to check and loop over them, performance should be alright even when there are 1000 parcels because it is done asynchronously.
A way to reduce the number of calls to the Ethereum node is to use `ethers` [JsonRpcBatchProvider](https://docs.ethers.org/v5/api/providers/other/#JsonRpcBatchProvider), in [this comment](https://github.com/ethers-io/ethers.js/issues/62#issuecomment-878558531) the author was wondering if 1 million calls is too much so 1000 shouldn't be a problem.

Reply:
Great find on the `JsonRpcBatchProvider`. We should try to limit RPC calls as much as possible. Looks like this is working well with Infura.

I also see Ceramic multiqueries working well and also the Ceramic caching.

This does now result in many IPFS lookups via our delegate node. This should be fine, as they are only calls for content routing and do not return or fetch actual content. However, there are many 404s to `/block/stat`. I will look into how to remove these calls, as they are unnecessary. `findprovs` is what is needed.